docs/examples/tensorflow/tensorflow-dataset-multigpu.ipynb

```diff
@@ -0,0 +1,320 @@
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "iterator_initializers = []\n",
+    "\n",
+    "with tf.device('/cpu:0'):\n",
+    "    tower_grads = []\n",
+    "\n",
+    "    for i in range(NUM_DEVICES):\n",
+    "        with tf.device('/gpu:{}'.format(i)):\n",
+    "            daliset = dali_tf.DALIDataset(\n",
+    "                pipeline=MnistPipeline(BATCH_SIZE, device_id=i, shard_id=i, num_shards=NUM_DEVICES),\n",
+    "                batch_size=BATCH_SIZE,\n",
+    "                shapes=shapes,\n",
+    "                dtypes=dtypes,\n",
+    "                device_id=i).with_options(options)\n",
+    "\n",
+    "            iterator = tf.data.make_initializable_iterator(daliset)\n",
+    "            iterator_initializers.append(iterator.initializer)\n",
+    "            images, labels = iterator.get_next()\n",
+    "\n",
+    "            images = tf.reshape(\n",
+    "                images, [BATCH_SIZE, IMAGE_SIZE*IMAGE_SIZE])\n",
+    "            labels = tf.reshape(\n",
+    "                tf.one_hot(labels, NUM_CLASSES),\n",
+    "                [BATCH_SIZE, NUM_CLASSES])\n",
+    "\n",
+    "            with tf.variable_scope('mnist_net', reuse=(i != 0)):\n",
+    "                images = tf.layers.flatten(images)\n",
+    "                images = tf.layers.dense(images, HIDDEN_SIZE, activation=tf.nn.relu)\n",
+    "                images = tf.layers.dropout(images, rate=DROPOUT, training=True)\n",
+    "                images = tf.layers.dense(images, NUM_CLASSES, activation=tf.nn.softmax)\n",
+    "\n",
+    "            logits_train = images\n",
+    "\n",
+    "            loss_op = tf.reduce_mean(tf.nn.softmax_cross_entropy_with_logits(\n",
+    "                logits=logits_train, labels=labels))\n",
+    "            optimizer = tf.train.AdamOptimizer()\n",
+    "            grads = optimizer.compute_gradients(loss_op)\n",
+    "\n",
+    "            if i == 0:\n",
+    "                correct_pred = tf.equal(\n",
+    "                    tf.argmax(logits_train, 1), tf.argmax(labels, 1))\n",
+    "                accuracy = tf.reduce_mean(\n",
+    "                    tf.cast(correct_pred, tf.float32))\n",
+    "\n",
+    "            tower_grads.append(grads)\n",
+    "\n",
+    "    tower_grads = average_gradients(tower_grads)\n",
+    "    train_step = optimizer.apply_gradients(tower_grads)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Everything is now ready to run the training. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Step 0, accuracy: 0.09375\n",
+      "Step 100, accuracy: 0.375\n",
+      "Step 200, accuracy: 0.4375\n",
+      "Step 300, accuracy: 0.546875\n",
+      "Step 400, accuracy: 0.421875\n",
+      "Final accuracy:  0.55296875\n"
```

Comment:
Something is wrong. It should be almost 100%

Reply:
I removed std value in CropMirrorNormalize by mistake, it's ok again.